### ISSUE TITLE
simplify And(Le|Eq(x, 3), Ge(x, 3)) to Eq(x, 3)

### ISSUE THREAD

**Issue by albertz** [sympy/sympy#18645]
```
x = Symbol("x")
```
With
```
expr = And(Ge(x, 3), Le(x, 3))
```
Or:
```
expr = And(Eq(x, 3), Le(x, 3))
```
I would expect that `expr.simplify()` yields `Eq(x, 3)` in both cases.
Maybe this is related to #18642 or #18389 but I'm not sure.

**Comment by asmeurer** [comment]
These sorts of assumptions based simplifications for the issues you opened generally aren't implemented yet. I'm not sure what a good general framework for doing them would be.

**Comment by oscarbenjamin** [comment]
The `solve` function can handle this:
```
In [10]: solve([Eq(x, 3), Le(x, 3)], x)                                                                                                        
Out[10]: x = 3
```
I don't think inequalities should be handled by solve (except as a filter on the outputs) but clearly some code exists to do this kind of thing.

**Comment by sylee957** [comment]
There is patternbased simplification for booleans. Maybe if this works, then it’s likely from there.

**Comment by oscargus** [comment]
There are patterns for this. However, the current master didn't use them for And-expressions at it had an early exit in case of no `Eq`. Fixed in #21854.

**Comment by oscarbenjamin** [comment]
I think linear programming (#21687) can handle these systems of inequalities.
